Code/Engine/NJSONArchives.h

```diff
@@ -0,0 +1,144 @@
+#pragma once
+
+#include <nlohmann/json.hpp>
+#include <entt/entt.hpp>
+
+// nlohmann-json based entt-(de)serialization
+class NJSONOutputArchive
+{
+public:
+    NJSONOutputArchive()
+    {
+        m_Root = nlohmann::json::array();
+    };
+
+    // new element for serialization. giving you the amount of elements that is going to be
+    // pumped into operator()(entt::entity ent) or operator()(entt::entity ent, const T &t)
+    void operator()(std::underlying_type_t<entt::entity> size)
+    {
+        int a = 0;
+        if (!m_Current.empty()) {
+            m_Root.push_back(m_Current);
+        }
+        m_Current = nlohmann::json::array();
+        m_Current.push_back(size); // first element of each array keeps the amount of elements. 
+    }
+
+    // persist entity ids
+    void operator()(entt::entity entity)
+    {
+        // Here it is assumed that no custom entt-type is chosen
+        m_Current.push_back((uint32_t)entity);
```

Comment:
Not sure if it would work to `static_cast` this, but we'd prefer it to c-style casting.

`m_Current.push_back(static_cast<uint32_t>(entity));`